app/src/widgets/ContactList/ui/ContactList.tsx

```diff
@@ -0,0 +1,175 @@
+import {
+    ChangeEvent,
+    FC,
+    useCallback,
+    useEffect,
+    useRef,
+    useState,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import { useNavigate } from 'react-router';
+import cn from 'classnames';
+import isUndefined from 'lodash/isUndefined';
+import SearchOutlined from '@ant-design/icons/SearchOutlined';
+
+import { Card, Input, InputVariants, LinearLoader } from 'ui-kit';
+
+import css from '@widgets/MessageList/ui/MessageList.module.scss';
+
+import {
+    useAddContactMutation,
+    useLazyGetAllContactsQuery,
+} from '@entities/contacts/api/slice';
+import { Contact } from '@entities/contacts/model/contact';
+
+import { FIND_USER_REGEX_MASK } from '@shared/config/regexMasks';
+import { ROUTE_CONFIG } from '@shared/config/routeConfig';
+import { useCounter } from '@shared/lib/hooks/useCounter';
+import { useDebounce } from '@shared/lib/hooks/useDebounce';
+import { useInfiniteScroll } from '@shared/lib/hooks/useInfiniteScroll';
+import { useIsToggled } from '@shared/lib/hooks/useIsToggled';
+
+const ContactList: FC = () => {
+    const [valueInput, setValueInput] = useState<string>('');
+    const [contacts, setContacts] = useState<Contact[]>([]);
+
+    const rootElement = useRef<HTMLDivElement | null>(null);
+    const isLoadingRef = useRef<boolean>(false);
+
+    const [getAllContacts, { isFetching, isError }] =
+        useLazyGetAllContactsQuery();
+
+    const [addContact, { isLoading: isLoadingAddContact }] =
+        useAddContactMutation();
+
+    const { t } = useTranslation();
+
+    const { isToggled, toggle } = useIsToggled();
+
+    const { count: page, set: setPage, increase } = useCounter(1);
+
+    const navigate = useNavigate();
+
+    const valueDebounce = useDebounce<string>(valueInput, 500);
+
+    const canNotAddContact =
+        !FIND_USER_REGEX_MASK.test(valueDebounce) ||
+        isFetching ||
+        isLoadingAddContact;
+
+    useEffect(() => {
+        isLoadingRef.current = isFetching;
```

Comment:
лучше дублировать для сохранения контекста 
```suggestion
        isFetchingRef.current = isFetching;
```